tools/detect-extension.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+function grepIniFile(string $filename, string $extension): int
+{
+    $lines = [];
+
+    foreach (new SplFileObject($filename) as $i => $line) {
+        if (strpos($line, 'extension') === false) {
+            continue;
+        }
+
+        if (strpos($line, $extension) === false) {
+            continue;
+        }
+
+        $lines[$i] = $line;
+    }
+
+    if (empty($lines)) {
+        printf("No interesting lines in %s.\n", $filename);
+
+        return 0;
+    }
+
+    printf("Interesting lines in %s...\n", $filename);
+    foreach ($lines as $i => $line) {
+        printf("  %d: %s\n", $i + 1, trim($line));
+    }
+
+    return count($lines);
+}
+
+$extension = $argv[1] ?? 'mongodb';
+$extensionDir = ini_get('extension_dir');
+
+$version = phpversion($extension);
+
+if (PHP_SAPI !== 'cli') {
+    echo '<pre>';
```

Comment:
Oh, I forgot I added this. It is notably omitted from `connect.php`, although I don't recall why that is.

Does it make sense to keep this here? Note that we never close this tag later, so we might be better off removing this and suggesting web SAPI users call this like `<pre><?php require(...); ?></pre>`.

Reply:
We can also close the `<pre>` tag before exiting - the missing closing tag doesn't matter when the script is run on its own, but could cause problems if the script was included in between other "stuff".

Reply:
I'm fine with either of these so long as we do the same for all scripts:

 * Wrap output in `<pre></pre>` when `PHP_SAPI` is not `cli`
 * Ignore any attempt for format the output and just tell users to use `<pre>` themselves in the docs

Slight preference for the second approach since it makes the scripts less complicated.

If you want to go with the first approach I think the cleanest way to do this would be something like the following so we don't need to worry about multiple exit points:

```php
if (PHP_SAPI !== 'cli') {
    echo '<pre>';
    register_shutdown_function(function() { echo '</pre>'; });
}
```

Reply:
I'm fine with either of these so long as we do the same for all scripts:

 * Wrap output in `<pre></pre>` when `PHP_SAPI` is not `cli`
 * Ignore any attempt for format the output and just tell users to use `<pre>` themselves in the docs

Slight preference for the second approach since it makes the scripts less complicated.

If you want to go with the first approach I think the cleanest way to do this would be something like the following so we don't need to worry about multiple exit points:

```php
if (PHP_SAPI !== 'cli') {
    echo '<pre>';
    register_shutdown_function(function() { echo '</pre>'; });
}
```

Reply:
> Using register_shutdown_function wouldn't echo the closing pre tag after returning from the corresponding script, but when the entire execution context finishes

Oh, good point there. This works for me.